apiserver/apiserver.go

```diff
@@ -846,7 +846,17 @@ func (srv *Server) endpoints() ([]apihttp.Endpoint, error) {
 			if err != nil {
 				return nil, nil, errors.Trace(err)
 			}
-			opener, err := resource.NewResourceOpener(st.State, store, srv.getResourceDownloadLimiter, tag.Id())
+			serviceFactory, err := httpCtxt.serviceFactoryForRequest(req)
```

Comment:
I think we have followed this function signature because model uuid has used the same signature all along. I feel as though this is far too liberal in what we are supplying. We really just want to supply the context on the request and not the request as well.

The defensive programmer wants to avoid unforeseen side effects giving the request to be potentially operated on.

Not a show stopper for this PR but I would be fixing this as it smells like bad passing a whole req when you only need a context. Also makes testing the logic easier and more correct.

Reply:
I'll do this in a new PR.